### PR TITLE
Remove legacy DDEBUG variable and cleanup win32.mak

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

**Review by Geod24 — APPROVED** [review]
LGTM but I'll let someone more Windows-savvy merge.

File: src/win32.mak [hunk 1]
```diff
@@ -117,33 +102,24 @@ check-host-dc:
 debdmd: check-host-dc debdmd-make
 
 debdmd-make:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest" "DOPT=" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_DEBUG=1" "DFLAGS=-unittest" $(TARGETEXE)
```
Comment (PetarKirov) [inline src/win32.mak:105]:
```suggestion
	$(DMDMAKE) "ENABLE_DEBUG=1" "ENABLE_UNITTEST=1" $(TARGETEXE)
```

File: src/win32.mak [hunk 2]
```diff
@@ -117,33 +102,24 @@ check-host-dc:
...
-	$(DMDMAKE) "DDEBUG=" "DOPT=-O -release -profile" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_PROFILE=1" "DFLAGS=-O -release" $(TARGETEXE)
 
-trace:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest" "DOPT=" $(TARGETEXE)
+trace: debdmd-make
 
 unittest:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest -cov" "DOPT=" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_DEBUG=1" "DFLAGS=-unittest -cov" $(TARGETEXE)
```
Comment (PetarKirov) [inline src/win32.mak:119]:
```suggestion
	$(DMDMAKE) "ENABLE_DEBUG=1" "ENABLE_UNITTEST=1" "DFLAGS=-cov" $(TARGETEXE)
```

An `ENABLE_COVERAGE` option would be useful here.

File: src/win32.mak [hunk 3]
```diff
@@ -117,33 +102,24 @@ check-host-dc:
...
+	$(RUN_BUILD) "ENABLE_RELEASE=1" $(TARGETEXE)
 
+# Not ENABLE_RELEASE to avoid -inline
 profile:
-	$(DMDMAKE) "DDEBUG=" "DOPT=-O -release -profile" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_PROFILE=1" "DFLAGS=-O -release" $(TARGETEXE)
 
-trace:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest" "DOPT=" $(TARGETEXE)
+trace: debdmd-make
```
Comment (PetarKirov) [inline src/win32.mak:116]: This target seems to be completely unnecessary, so I'd be okay with removing it.

**Review by PetarKirov — COMMENTED** [review]
LGTM too, module a some minor comments:

File: src/win32.mak (hunk 1 above)
Comment (MoonlightSentinel) [reply src/win32.mak]: Fixed

File: src/win32.mak (hunk 2 above)
Comment (MoonlightSentinel) [reply src/win32.mak]: Fixed. `ENABLE_COVERAGE` exists already.

File: src/win32.mak (hunk 3 above)
Comment (MoonlightSentinel) [reply src/win32.mak:116]: It's probably unused but I don't mind keeping it as a simple alias.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**